src/server/project.ts

```diff
@@ -878,13 +875,34 @@ namespace ts.server {
                 return;
             }
 
+            // Search our peer node_modules, then any globally-specified probe paths
+            // ../../.. to walk from X/node_modules/typescript/lib/tsserver.js to X/node_modules/
+            const searchPaths = [combinePaths(host.getExecutingFilePath(), "../../.."), ...this.projectService.pluginProbeLocations];
+
+            // Enable tsconfig-specified plugins
             for (const pluginConfigEntry of options.plugins) {
-                const searchPath = getDirectoryPath(this.configFileName);
-                const resolvedModule = <PluginModuleFactory>Project.resolveModule(pluginConfigEntry.name, searchPath, host, log);
+                this.enablePlugin(pluginConfigEntry, searchPaths);
+            }
+            // Enable global plugins with synthetic configuration entries
+            for (const globalPluginName of this.projectService.globalPlugins) {
+                // Provide global: true so plugins can detect why they can't find their config
+                this.enablePlugin({ name: globalPluginName, global: true } as PluginImport, searchPaths);

```

Comment:
Should we skip loading a global if a local copy of the same plugin is given (as that can have config also).